tests/unit/httpd_test.cc

```diff
@@ -205,6 +205,24 @@ SEASTAR_TEST_CASE(test_decode_url) {
     return make_ready_future<>();
 }
 
+SEASTAR_TEST_CASE(test_decode_path) {
```

Comment:
I would say that's not enough. There's a `test_json_path` test that validates how parameters matching works, I think _that_ test should be extended to validate the new change

Reply:
Thanks, this made me realise that the path matcher adds arguments into the internal map of params with a ‘/‘ prefix, so I fixed get_decoded_param to remove this from the return value to match the behaviour of operator[].

I've updated the `test_json_path` unit test, let me know if there's anything else you had in mind here.

Reply:
Please take a look at the tests I wrote in https://github.com/scylladb/seastar/pull/2125 and consider adopting them (they are almost trivial, I think there's one case your test doesn't cover).
It's "too easy" to write a super-low-level unit test that calls your new function and checks it does what you wrote in it. It's more interesting to use encoded parameters in the higher-up APIs and see that the correct thing happens to them.
I also have a very-high-level test in ScyllaDB that uses the API and checks it works as it should, https://github.com/scylladb/scylladb/pull/18091, but that will be a test that I will run later - it can't be included in this PR obviously. I hope you also have your own high-level tests in Redpanda which reproduced the original bug and show that your fix is correct.